apps/server/src/modules/instagram-auth/services/instagram-auth.service.ts

```diff
@@ -31,19 +27,30 @@ export class InstagramAuthService {
 			throw new BadRequestException('User not found');
 		}
 
-		const existingAccount = await this.userModel.countDocuments({
-			_id: user?._id,
-			'InstagramAccounts.username': username,
-		});
+		const existingAccount = await this.userModel
+			.findOne(
+				{
+					_id: user?._id,
```

Comment:
```suggestion
					_id: user._id,
```

ja validou acima se nao existe